modules/apps/marketplace/marketplace-js-components-web/test/js/core/MarketplaceRest.spec.ts

```diff
@@ -0,0 +1,362 @@
+			globalThis.Liferay.Util.SessionStorage.TYPES.NECESSARY
+		);
+
+		expect(marketplaceAuthorization).toMatchObject(marketplaceToken);
+	});
+
+	it('fetch marketplace token from cache', async () => {
+		const marketplaceAuthorization =
+			await marketplaceRest.getMarketplaceToken();
+
+		expect(fetch).toBeCalledTimes(0);
+		expect(getItemFn).toBeCalledTimes(1);
+		expect(setItemFn).toBeCalledTimes(0);
+		expect(marketplaceAuthorization).toMatchObject(marketplaceToken);
+	});
+
+	it('fetch marketplace products', async () => {
+		const {fetch} = require('frontend-js-web');
+
+		const productResponse = {
+			items: [],
+			page: 1,
+			pageSize: 20,
+			totalCount: 0,
+		};
+
+		fetch.mockResponseOnce(JSON.stringify(productResponse));
+
+		const response = await marketplaceRest.getProducts();
+
+		const [fetchURL, fetchParams] = fetch.mock.calls[0];
+
+		expect(fetch).toBeCalledTimes(1);
+		expect(fetchParams.guestOperation).toBe(true);
+		expect(fetchURL).toBe(
+			`${marketplaceConfiguration.url}/o/headless-commerce-delivery-catalog/v1.0/channels/${marketplaceConfiguration.settings.channelId}/products?`
+		);
+		expect(response).toMatchObject(productResponse);
+	});
+
+	it('fetch marketplace service', async () => {
+		const {fetch} = require('frontend-js-web');
+
+		const mockResponse = {success: true};
+
+		fetch.mockResponseOnce(JSON.stringify(mockResponse));
+
+		const response =
+			await marketplaceRest.fetchMarketplaceService('/dxp/test');
+
+		const [fetchURL] = fetch.mock.calls[0];
+
+		expect(fetchURL).toBe(
+			`${marketplaceConfiguration.serviceURL}/dxp/test`
+		);
+
+		expect(fetch).toBeCalledTimes(1);
+
+		expect(response).toMatchObject(mockResponse);
+	});
+
+	it('fetch console provisioning', async () => {
+		const {fetch} = require('frontend-js-web');
+
+		const cart = {
+			cartItems: [{id: 456}],
+			id: 123,
+		} as Cart;
+
+		const mockResponse = {success: true};
+
+		fetch.mockResponseOnce(JSON.stringify(mockResponse));
+
+		await marketplaceRest.consoleProvisioningOrder(cart);
+
+		const [fetchURL, fetchParams] = fetch.mock.calls[0];
+
+		expect(fetchURL).toBe(
+			`${marketplaceConfiguration.serviceURL}/dxp/provisioning/${cart.id}`
+		);
+		expect(fetchParams.body).toMatch(
+			JSON.stringify({
+				orderItemId: 456,
+				projectId: marketplaceConfiguration.settings.cloudProject,
+			})
+		);
+		expect(fetchParams.method).toBe('POST');
+	});
+
+	it('fetch base resource url', async () => {
+		const baseURL = MarketplaceRest.getBaseResourceURL();
+		expect(baseURL).toBe(
+			`/group/guest/~/control_panel/manage?p_p_id=${globalThis.Liferay.PortletKeys.INSTANCE_SETTINGS}`
+		);
+	});
+
+	it('fetch Marketplace without guest operation', async () => {
+		const {fetch} = require('frontend-js-web');
+
+		const mockResponse = {success: true};
+
+		fetch.mockResponseOnce(JSON.stringify(mockResponse));
+
+		const response = await marketplaceRest.fetchMarketplace('/dxp/test');
+
+		const [fetchURL, fetchCall] = fetch.mock.calls[0];
```

Comment:
can you sort it? and if possible remove the blank line.
Try to remove blank line and run format source, if it works please apply to the others